rust/datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -324,12 +362,24 @@ impl GroupedHashAggregateStream {
         aggr_expr: Vec<Arc<dyn AggregateExpr>>,
         input: SendableRecordBatchStream,
     ) -> Self {
+        let (tx, rx) = futures::channel::oneshot::channel();
+
+        let schema_clone = schema.clone();
+        tokio::spawn(async move {
```

Comment:
this spawns a separate task to do the actual aggregation (and drive the inputs, as needed, by calling `input.next().await`). Driving the input was not happening when using `future.into_stream` in the actual call to `poll_next` https://github.com/apache/arrow/pull/8553/files#diff-4dbe750a2836f6c723bc36ae39f2faf15e2706303f5dbb042ae00f1f36658c62L399